src/main/java/org/motechproject/prevac/service/impl/VisitRescheduleServiceImpl.java

```diff
@@ -146,14 +149,6 @@ private void checkNumberOfPatients(VisitRescheduleDto dto, Clinic clinic) { //NO
     }
 
     private void validateDate(VisitRescheduleDto dto, Visit visit) {
-        if (visit.getDate() != null) {
-            throw new IllegalArgumentException("Cannot reschedule, because Visit already took place");
-        }
-
-        if (dto.getPlannedDate().isBefore(LocalDate.now())) {
-            throw new IllegalArgumentException("Date cannot be in the past");
-        }
-
         if (!dto.getIgnoreDateLimitation()) {
```

Comment:
I think we should change the min and max dates after calculation (depending if actual or planned date is validated) to not allow planned dates in the past and actual dates in future.